src/R/create_deprivation_index.R

```diff
@@ -4,37 +4,59 @@ library(maptools)
-
-proj4string(milano_shp) = CRS("+proj=utm +zone=32 +ellps=WGS84 +datum=WGS84 +units=m +no_defs ")
-milano_shp = spTransform(milano_shp, CRS("+init=epsg:4326"))
-# proj4string(lombardia_shp) = CRS("+proj=utm +zone=32 +ellps=WGS84 +datum=WGS84 +units=m +no_defs ")
-# lombardia_shp = spTransform(lombardia_shp, CRS("+init=epsg:4326"))
-# lombardia_shp@data = lombardia_shp@data %>% 
-#   mutate(SEZ2011 = as.character(SEZ2011)) %>%
-#   left_join(census, by = c("SEZ2011"))
 
+granularity_level = "Census Area" ## Choose 'Block' or 'Census Area'
+# region = "Italy" ## Choose 'Milano', 'Trento' or 'Milan'
 
+# census = multmerge("data/census/Sezioni di Censimento", "R\\d", read_census_data)
+census = multmerge("data/census/Aree di Censimento", "R\\d", function(x){read_delim(file=x, delim = ";", na = "null")})
+
+milano_sez = readShapePoly("data/milan_sez_2011/Milano_SezC_2011.shp")
+proj4string(milano_sez) = CRS("+proj=utm +zone=32 +ellps=WGS84 +datum=WGS84 +units=m +no_defs")
+milano_sez = spTransform(milano_sez, CRS("+init=epsg:4326"))
+
+milano_ace = readShapePoly("data/milan_ace_2011/ACE_maggio_2011.shp")
+proj4string(milano_ace) = CRS("+proj=tmerc +lat_0=0 +lon_0=9 +k=0.9996 +x_0=1500000 +y_0=0 
+                          +ellps=intl +units=m +no_defs ")
+milano_ace = spTransform(milano_ace, CRS("+init=epsg:4326"))
+
+
+# italy_sez = readShapePoly("data/Italy_SezC_2011/Italy_SezC_2011.shp")
+# proj4string(italy) = CRS("+proj=utm +zone=32 +ellps=WGS84 +datum=WGS84 +units=m +no_defs ")
+# italy = spTransform(italy, CRS("+init=epsg:4326"))
 
-#### Build Deprivation Index using PCA 
-## Read all census dataset
-# census = multmerge("data/dati-cpa_2011_/Aree di Censimento", "R\\d")
 
+
+#### Build Deprivation Index using PCA 
 census_depriv = census %>% 
-  transmute(P47 = P47/P1, P48 = P48/P1, P49 = P49/P1, P50 = P50/P1, P51 = P51/P1, 
-            P52 = P52/P1, P27 = P27/P1, P28 = P28/P1, P29 = P29/P1, ST15 = ST15/P1,
-            A46 = ifelse(A2 > 0, A46/A2, NA), A47 = ifelse(A2 > 0, A47/A2, NA), 
-            P62 = P62/P60, P61 = P61/P60, P60 = P60/P1, E21 = E21/E3, E22 = E22/E3, 
-            E23 = E23/E3, E24 = E24/E3, E25 = E25/E3,E26 = E26/E3) 
-row.names(census_depriv) = census$SEZ2011
+  transmute(high_school = P48/P1, 
+            illiteracy = P52/P1, sixtyfive_plus = (P27 + P28 + P29)/P1,
+            foreigners = ST15/P1,
+            rented_dwelling = ifelse(A46 + A47+ A48 > 0, A46/(A46 + A47+ A48), NA),
+            unemployment = P62/P60,
+            work_force = P60/(P17 + P18 + P19 + P20 + P21 + P22 + P23 + P24 + P25 + P26 + P27 + P28 + P29)
+            ) 
+
+if(granularity_level == "Block"){
+  row.names(census_depriv) = census$SEZ2011
+}else if(granularity_level == "Census Area"){
+  row.names(census_depriv) -> census$ACE
+  }
+
+
+
+png("corr_plot.png", height = 1000, width = 1000)
+corr_plot = ggpairs(census_depriv, lower = list(continuous = wrap("points", alpha = 0.15)))
```

Comment:
Does this create a pairwise correlation of every two variables? I'd like to see that!